python_modules/libraries/dagster-tableau/dagster_tableau/assets.py

```diff
@@ -0,0 +1,105 @@
+from typing import Optional, Sequence, cast
+
+from dagster import (
+    AssetExecutionContext,
+    AssetsDefinition,
+    AssetSpec,
+    ObserveResult,
+    Output,
+    _check as check,
+    multi_asset,
+)
+
+from dagster_tableau.resources import (
+    BaseTableauWorkspace,
+    TableauCloudWorkspace,
+    TableauServerWorkspace,
+)
+
+
+def build_tableau_executable_assets_definition(
+    resource_key: str,
+    workspace: BaseTableauWorkspace,
+    specs: Sequence[AssetSpec],
+    refreshable_workbook_ids: Optional[Sequence[str]] = None,
+) -> AssetsDefinition:
+    """Returns the AssetsDefinition of the executable assets in the Tableau workspace.
+
+    Args:
+        resource_key (str): The resource key to use for the Tableau resource.
+        workspace (BaseTableauWorkspace): The Tableau workspace to fetch assets from.
+        specs (Sequence[AssetSpec]): The asset specs of the executable assets in the Tableau workspace.
+        refreshable_workbook_ids (Optional[Sequence[str]]): A list of workbook IDs. The workbooks provided must
+            have extracts as data sources and be refreshable in Tableau.
+
+            When materializing your Tableau assets, the workbooks provided are refreshed,
+            refreshing their sheets and dashboards before pulling their data in Dagster.
+
+            This feature is equivalent to selecting Refreshing Extracts for a workbook in Tableau UI
+            and only works for workbooks for which the data sources are extracts.
+            See https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_workbooks_and_views.htm#update_workbook_now
+            for documentation.
+
+    Returns:
+        AssetsDefinition: The AssetsDefinition of the executable assets in the Tableau workspace.
+    """
+
+    @multi_asset(
+        name=f"tableau_sync_site_{workspace.site_name.replace('-', '_')}",
```

Comment:
can we use the resource key here, instead? that should be unique

Reply:
Done in https://github.com/dagster-io/dagster/pull/25459/commits/7601ab3235bfd30aeb48361b5034730e3e88dc5f